src/types/navigation.ts

```diff
@@ -0,0 +1,29 @@
+import { TIcon } from '@components';
+
+import { TURL } from './url';
+
+export interface IRouteLink {
+  type: 'internal';
+  title: string;
+  to: string;
+  enabled: boolean;
+  icon: TIcon;
+}
+
+export interface IExternalLink {
+  type: 'external';
+  title: string;
+  link: TURL;
+  analyticsEvent: string;
+  icon: TIcon;
+}
+
+export type TTrayItem = IRouteLink | IExternalLink;
```

Comment:
I'm wondering if we can save ourself the typings.
Can TS infer these types ? Have you considered creating a common interface for both ?

Reply:
TS can't infer these types and I need those on Component props. I've considered using a common interface for both but it adds a lot of conditional on components and sometimes it's not even usable.

Using discriminated unions was the cleanest way to define those